master/internal/core_checkpoint.go

```diff
@@ -127,6 +152,50 @@ func (m *Master) getCheckpointImpl(
 	return nil
 }
 
+// @Summary Check for access to a checkpoint's files for download.
+// @Tags Checkpoints
+// @ID check-checkpoint
+// @Accept  json
+// @Produce  json
+// @Param   checkpoint_uuid path string  true  "Checkpoint UUID"
+// @Success 200 {} string ""
+//nolint:godot
+// @Router /checkpoints/{checkpoint_uuid}/access [get]
+func (m *Master) checkCheckpoint(c echo.Context) error {
+	args := struct {
+		CheckpointUUID string `path:"checkpoint_uuid"`
+	}{}
+	if err := api.BindArgs(&args, c); err != nil {
+		return echo.NewHTTPError(http.StatusBadRequest, "invalid checkpoint_uuid: "+err.Error())
+	}
+	id, err := uuid.Parse(args.CheckpointUUID)
+	if err != nil {
+		return echo.NewHTTPError(http.StatusBadRequest,
+			fmt.Sprintf("unable to parse checkpoint UUID %s: %s",
+				args.CheckpointUUID, err))
+	}
+
+	curUser := c.(*detContext.DetContext).MustGetUser()
+	if err := m.canDoActionOnCheckpoint(c.Request().Context(), curUser, args.CheckpointUUID,
+		expauth.AuthZProvider.Get().CanGetExperimentArtifacts); err != nil {
+		s, ok := status.FromError(err)
+		if !ok {
+			return err
+		}
+		switch s.Code() {
+		case codes.NotFound:
+			return echo.NewHTTPError(http.StatusNotFound, s.Message())
+		case codes.PermissionDenied:
+			return echo.NewHTTPError(http.StatusForbidden, s.Message())
+		default:
+			return fmt.Errorf(s.Message())
+		}
+	}
```

Comment:
this is identical to the code in `getCheckpoint`. can we please refactor it out into a helper function?